tgui/packages/tgui/interfaces/Microscope.tsx

```diff
@@ -0,0 +1,183 @@
+import { BooleanLike } from 'common/react';
+
+import { useBackend } from '../backend';
+import {
+  Box,
+  Button,
+  Divider,
+  DmIcon,
+  Icon,
+  NoticeBox,
+  Section,
+  Stack,
+} from '../components';
+import { Window } from '../layouts';
+
+type Data = {
+  has_dish: BooleanLike;
+  cell_lines: CellLine[];
+};
+
+type CellLine = {
+  type: string;
+  name: string;
+  desc: string;
+  icon: string;
+  icon_state: string;
+  consumption_rate: number;
+  growth_rate: number;
+  suspectibility: number;
+  requireds: Reagent[];
+  supplementaries: Reagent[];
+  suppressives: Reagent[];
+};
+
+type Reagent = {
+  [key: string]: number;
+};
+
+export const Microscope = (props) => {
+  const { act, data } = useBackend<Data>();
+  const { has_dish, cell_lines = [] } = data;
+
+  return (
+    <Window width={620} height={620}>
+      <Window.Content scrollable>
+        <Section
+          title={has_dish ? 'Petri Dish Sample' : 'No Petri Dish'}
+          buttons={
+            !!has_dish && (
+              <Button
+                icon="eject"
+                disabled={!has_dish}
+                onClick={() => act('eject_petridish')}
+              >
+                Take Dish
+              </Button>
+            )
+          }
+        >
+          <CellList cell_lines={cell_lines} />
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const CellList = (props) => {
+  const { cell_lines } = props;
+  if (!cell_lines.length) {
+    return <NoticeBox>No micro-organisms found</NoticeBox>;
+  }
+
+  return cell_lines.map((cell_line) => {
+    return cell_line.type !== 'virus' ? (
+      <Stack key={cell_line.desc} mt={2}>
+        <Stack.Item>
+          <DmIcon
+            icon={cell_line.icon}
+            icon_state={cell_line.icon_state}
+            height="64px"
+            width="64px"
+          />
+        </Stack.Item>
+        <Stack.Item grow>
+          <Section
+            title={cell_line.desc}
+            buttons={
+              <Button
+                color="transparent"
+                icon="circle-question"
+                tooltip="Put the sample into a Growing Vat and pour the required reagents."
+              />
+            }
+          >
+            <Box my={1}>
+              Consume {cell_line.consumption_rate} units of every nutrient per
+              second to grow by {cell_line.growth_rate}%.
+            </Box>
+            {cell_line.suspectibility > 0 && (
+              <Box my={1}>
+                Reduced by {cell_line.suspectibility}% when infected with
+                viruses.
+              </Box>
+            )}
+            <Stack fill>
+              <Stack.Item grow>
+                <GroupTitle title="Required Reagents" />
+                {Object.keys(cell_line.requireds).map((reagent) => (
+                  <Button fluid key={reagent}>
```

Comment:
is reagent an object? if so, please no objects as keys. even the index would be better here

https://react.dev/learn/rendering-lists#rendering-data-from-arrays

Reply:
It's a reagent name.